instrumentation/spring/spring-web-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springweb/WebApplicationContextInstrumentation.java

```diff
@@ -65,14 +65,16 @@ public static void onEnter(@Advice.Argument(0) ConfigurableListableBeanFactory b
           // Firstly check whether DispatcherServlet is present. We need to load an instrumented
           // class from spring-webmvc to trigger injection that makes
           // OpenTelemetryHandlerMappingFilter available.
-          beanFactory
-              .getBeanClassLoader()
-              .loadClass("org.springframework.web.servlet.DispatcherServlet");
-
-          // Now attempt to load our injected instrumentation class.
-          Class<?> clazz =
+          Class<?> dispatcherServletClass =
               beanFactory
                   .getBeanClassLoader()
+                  .loadClass("org.springframework.web.servlet.DispatcherServlet");
+
+          // Now attempt to load our injected instrumentation class from the same class loader as
+          // DispatcherServlet
+          Class<?> clazz =
```

Comment:
We have a test with wildfly where loading our injected class from `getBeanClassLoader()` fails. Previously it worked because it hit `findLoadedClass` during class loading.
If we do it this way we can probably put `OpenTelemetryHandlerMappingFilter` back into our own package. We needed to put in spring package because we set up the bean definition in the wrong way. We set both name and class that are actually kept in the same field in spring bean definition so it always tried to use loadClass which previously worked only if the class was in spring package.